osu.Game/Overlays/Chat/ReportChatPopover.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Game.Graphics.UserInterfaceV2;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Online.Chat;
+using osu.Game.Resources.Localisation.Web;
+
+namespace osu.Game.Overlays.Chat
+{
+    public partial class ReportChatPopover : ReportPopover<ChatReportReason>
+    {
+        [Resolved]
+        private IAPIProvider api { get; set; } = null!;
+
+        [Resolved]
+        private ChannelManager channelManager { get; set; } = null!;
+
+        private readonly Message message;
+
+        public ReportChatPopover(Message message)
+            : base(ReportStrings.UserTitle(message.Sender?.Username ?? @"Someone"))
+        {
+            this.message = message;
+            Action = report;
+        }
+
+        protected override bool CheckCanSubmitEmptyComment(ChatReportReason reason)
+        {
+            return reason != ChatReportReason.Other;
+        }
+
+        private void report(ChatReportReason reason, string comments)
+        {
+            var request = new ChatReportRequest(message.Id, reason, comments);
+
+            request.Success += () => Schedule(() =>
+            {
+                channelManager.CurrentChannel.Value.AddNewMessages(new InfoMessage(UsersStrings.ReportThanks.ToString()));
+            });
```

Comment:
And whatever this thing is, it just... doesn't work? The message addition is scheduled to a popover that is no longer present, and is going to be disposed. What is going on here?

It doesn't even work _in the test_...

Reply:
That may be on me as part of my final refactoring to avoid the reporting flow sitting in the wrong place.

I switched this to using a notification for feedback then flip-flopped back to this because it felt like better UX.

Reply:
I'm not really sure why this was ever scheduled but judging from stuff such as

https://github.com/ppy/osu/blob/8c3b1faa01f7e3463d016cb27ff4687ef94c3844/osu.Game/Online/Chat/ChannelManager.cs#L376-L380

it doesn't look like it really *needs* to be, so I guess I'll just get rid of the schedule and call it a day.

Reply:
It seems that removing `Schedule` will turn normal